src/main/java/net/neoforged/neoforge/registries/DeferredRegister.java

```diff
@@ -331,60 +279,294 @@ public ResourceKey<? extends Registry<T>> getRegistryKey() {
+        }
+
+        /**
+         * Adds a new block to the list of entries to be registered and returns a {@link DeferredHolder} that will be populated with the created block automatically.
+         *
+         * @param name The new block's name. It will automatically have the {@linkplain #getNamespace() namespace} prefixed.
+         * @param sup  A factory for the new block. The factory should not cache the created block.
+         * @return A {@link DeferredHolder} that will track updates from the registry for this block.
+         */
+        @Override
+        public <B extends Block> DeferredBlock<B> register(String name, Supplier<? extends B> sup) {
+            return this.register(name, key -> sup.get());
+        }
+
+        /**
+         * Adds a new block to the list of entries to be registered and returns a {@link DeferredHolder} that will be populated with the created block automatically.
+         *
+         * @param name  The new block's name. It will automatically have the {@linkplain #getNamespace() namespace} prefixed.
+         * @param func  A factory for the new block. The factory should not cache the created block.
+         * @param props The properties for the created block.
+         * @return A {@link DeferredHolder} that will track updates from the registry for this block.
+         * @see #registerBlock(String, BlockBehaviour.Properties)
+         */
+        public <B extends Block> DeferredBlock<B> registerBlock(String name, Function<BlockBehaviour.Properties, ? extends B> func, BlockBehaviour.Properties props) {
+            return this.register(name, () -> func.apply(props));
+        }
+
+        /**
+         * Adds a new block to the list of entries to be registered and returns a {@link DeferredHolder} that will be populated with the created block automatically.
+         *
+         * @param name  The new block's name. It will automatically have the {@linkplain #getNamespace() namespace} prefixed.
+         * @param props The properties for the created block.
+         * @return A {@link DeferredHolder} that will track updates from the registry for this block.
+         * @see #registerBlock(String, Function, BlockBehaviour.Properties)
+         */
+        public DeferredBlock<Block> registerBlock(String name, BlockBehaviour.Properties props) {
+            return this.registerBlock(name, Block::new, props);
+        }
+
+        @Override
+        protected <I extends Block> DeferredBlock<I> createHolder(ResourceKey<? extends Registry<Block>> registryKey, ResourceLocation key) {
+            return DeferredBlock.createBlock(ResourceKey.create(registryKey, key));
         }
     }
 
-    private void createRegistry(NewRegistryEvent event) {
-        event.create(this.registryFactory.get(), this::onFill);
+    /**
+     * Specialized DeferredRegister for {@link Item Items} that uses the specialized {@link DeferredHolder}, {@link DeferredItem} as the return type for {@link #register}.
```

Comment:
The reference to `DeferredHolder` is unnecessary and makes the javadoc less readable in my eyes.
```suggestion
     * Specialized DeferredRegister for {@link Item Items} that uses the specialized {@link DeferredItem} as the return type for {@link #register}.
```

Reply:
Fixed in f2801cb37.